File: Answers/Answer_6/normal_distribution_writer.py

import ROOT
from array import array

N = 1000

out_file = ROOT.TFile.Open("normal_distribution.root", "RECREATE")

normal_distribution = ROOT.TF1("normal_distribution", "[0]*exp(-0.5*((x-[1])/[2])**2)", -100, 100)
normal_distribution.SetParameter(0, 200)
normal_distribution.SetParameter(1, 0)
normal_distribution.SetParameter(2, 1)

t1 = ROOT.TTree("t1", "normal distribution")
normal_value = array("f", [0.0])
t1.Branch("normal_value", normal_value, "normal_value/F")

for i in range(N):
    normal_value[0] = normal_distribution.GetRandom()
    t1.Fill()

t1.Write()
out_file.Close()

print("\"normal_distribution.root\" successfully written.")

File: Answers/Answer_6/normal_distribution_writer.cpp

#include <iostream>
#include "TF1.h"
#include "TTree.h"
#include "TFile.h"
using namespace std;

int main() {

    int N = 1000;

    TFile out_file("normal_distribution.root","RECREATE");

    auto normal_distribution = new TF1("normal_distribution", "[0]*exp(-0.5*((x-[1])/[2])**2)", -100, 100);
    normal_distribution->SetParameter(0, 200);
    normal_distribution->SetParameter(1, 0);
    normal_distribution->SetParameter(2, 1);
    
    TTree t1("t1", "normal distribution");
    float normal_value;
    t1.Branch("normal_value", &normal_value, "normal_value/F");

    for (int i=0;i<N;i++) {
        normal_value = normal_distribution->GetRandom();
        t1.Fill();
    }

    t1.Write();
    out_file.Close();

    cout << "\"normal_distribution.root\" successfully written." << endl;
    
    return 0;

}
